### PR TITLE
fix: Fix scanning google cloud with service account credentials file

### PR DESCRIPTION
**Pull request by nameexhaustion**
Fixes https://github.com/pola-rs/polars/issues/19781

### CODE REVIEW

File: py-polars/polars/io/cloud/credential_provider.py [hunk 1]
```diff
@@ -168,7 +181,16 @@ def __init__(self) -> None:
         #
         # So we just bypass it with a __dict__[] (because ruff complains about
         # getattr) :|
-        creds, _ = google.auth.__dict__["default"]()
+        creds, _ = google.auth.__dict__["default"](
+            scopes=(
+                scopes
+                if scopes is not None
+                else ["https://www.googleapis.com/auth/cloud-platform"]
```
Comment (nameexhaustion) [inline py-polars/polars/io/cloud/credential_provider.py:188]: We need to pass this URL into `scopes` - https://stackoverflow.com/a/74562563

File: py-polars/polars/meta/versions.py [hunk 2]
```diff
@@ -68,12 +68,14 @@ def _get_dependency_list() -> list[str]:
...
         "adbc_driver_manager",
         "altair",
+        "boto3",
         "cloudpickle",
         "connectorx",
         "deltalake",
         "fastexcel",
         "fsspec",
         "gevent",
+        "google.auth",
```
Comment (nameexhaustion) [inline py-polars/polars/meta/versions.py:78]: drive-by print the `boto3` and `google-auth` package versions in `show_versions()`, since we could be using them now

**Review by ritchie46 — APPROVED** [review]
